crates/sui-framework/sources/vec_map.move

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::vec_map {
+    use std::option::{Self, Option};
+    use std::vector;
+
+    /// This key already exists in the map
+    const EKeyAlreadyExists: u64 = 0;
+
+    /// This key does not exist in the map
+    const EKeyDoesNotExist: u64 = 1;
+
+    /// Trying to destroy a map that is not empty
+    const EMapNotEmpty: u64 = 2;
+
+    /// Trying to access an element of the map at an invalid index
+    const EIndexOutOfBounds: u64 = 3;
+
+    /// A map data structure backed by a vector. The map is guaranteed not to contain duplicate keys, but entries
+    /// are *not* sorted by key--entries are included in insertion order.
+    /// All operations are O(N) in the size of the map--the intention of this data structure is only to provide
+    /// the convenience of programming against a map API.
+    /// Large maps should use the `Collection` or `Bag` type instead.
```

Comment:
Collection/Bag isn't more efficient (time complexity wise) either. It also uses a vector of keys much like VecMap. In fact, with this change, I would probably modify Collection/Bag to use VecMap.

Reply:
Good point! I will modify this comment.